Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,356 @@
+%             pair-wise subtraction (boolean, REQUIRED)
+%             NB: see xASL_quant_HadamardDecoding for fields required by
+%             time encoding
+%   x.Q     - field containing quantification parameters (REQUIRED)
+%   x.Q.EchoTime - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+%   x.Q.Initial_PLD - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+%   x.Q.LabelingDuration - numerical vector. If this is a scalar, it will be extended to the number of image volumes (REQUIRED)
+% 
+%   path_ASL4D - Path to the encoded ASL4D image we want to decode (STRING OR 4D MATRIX, REQUIRED)
+%             Alternatively, this can contain the image matrix
+%             (xASL_io_Nifti2Im below allows both path and image matrix inputs).
+%             Best if this is a path, so any motion correction can be applied.
+%   PWI4D   - likewise, see explanation below. If this is provided, it will override the PWI4D calculation below (STRING OR 4D MATRIX, OPTIONAL,
+%               DEFAULT = generated below from ASL4D)
+%   PWI3D   - likewise, see explanation below. If this is provided, it will override the PWI4D calculation below (STRING OR 4D MATRIX, OPTIONAL,
+%               DEFAULT = generated below from PWI4D)
+%
+% OUTPUT:
+%   PWI     - 3D image matrix, see explanation below
+%   PWI3D   - 4D image matrix, see explanation below
+%   PWI4D   - 4D image matrix, see explanation below
+%   x       - updated structure, with the following quantification parameters:
+%             fitting to the nomenclature explained below 
+%
+%   x.Q.EchoTimeLabelingDuration_PWI4D
+%   x.Q.Initial_PLD_PWI4D
+%   x.Q.LabelingDuration_PWI4D
+%
+%   x.Q.EchoTimeLabelingDuration_PWI3D
+%   x.Q.Initial_PLD_PWI3D
+%   x.Q.LabelingDuration_PWI3D
+%
+%   x.Q.EchoTimeLabelingDuration_PWI
+%   x.Q.Initial_PLD_PWI
+%   x.Q.LabelingDuration_PWI
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DEVELOPER PM:
+% 
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function performs subtraction and averaging all across the ExploreASL ASL modules, such as xASL_wrp_Realign,
+%              and xASL_wrp_RegisterASL to compute temporary PWI images, xASL_wrp_ResampleASL for creating QC images and preparing
+%              quantification, and xASL_wrp_Quantify if PWI4D is not used by the quantification algorithm (such as BASIL)
+%
+% This function performs the following steps:
+% 1. Admin & checks
+% 2. Apply motion correction if this is needed
+%              
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes
```

Comment:
I know, this is not ideal. This is just historically grown because we already had PWI4D and PWI and we needed something in between. I suggest to now keep it like this and change this in a future issue. With BIDS derivatives we will probably do something like (left legacy, right BIDS):

ASL4D -> asl
PWI4D -> asl_subtracted
PWI3D -> asl_subtractedAveraged
PWI      -> asl_PWI

Reply:
sure. that works.